rtnetlink/src/neighbour/add.rs

```diff
@@ -0,0 +1,85 @@
+use futures::stream::StreamExt;
+
+use netlink_packet_route::{
+    constants::*,
+    neighbour::{NeighbourMessage, Nla},
+    NetlinkPayload,
+    RtnlMessage,
+};
+
+use netlink_proto::packet::NetlinkMessage;
+
+use crate::{Error, Handle};
+use std::net::IpAddr;
+
+pub struct NeighbourAddRequest {
+    handle: Handle,
+    message: NeighbourMessage,
+}
+
+impl NeighbourAddRequest {
+    pub(crate) fn new(handle: Handle, index: u32, destination: IpAddr) -> Self {
+        let mut message = NeighbourMessage::default();
+
+        message.header.family = match destination {
+            IpAddr::V4(_) => AF_INET as u8,
+            IpAddr::V6(_) => AF_INET6 as u8,
+        };
+
+        message.header.ifindex = index;
+        message.header.state = IFA_F_PERMANENT as u16;
+        message.header.ntype = NDA_UNSPEC as u8;
+
+        message.nlas.push(Nla::Destination(match destination {
+            IpAddr::V4(v4) => v4.octets().to_vec(),
+            IpAddr::V6(v6) => v6.octets().to_vec(),
+        }));
+
+        NeighbourAddRequest { handle, message }
+    }
+
+    pub fn state(mut self, state: u16) -> Self {
```

Comment:
We should explain what the `state`, `flags` and `ntype` are about. This is already documented a little in `netlink-packet-route`: https://docs.rs/netlink-packet-route/0.6.0/netlink_packet_route/rtnl/struct.NeighbourHeader.html.